tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -201,6 +203,8 @@ void SPIRVEmitter::doDecl(const Decl *decl) {
     doVarDecl(varDecl);
   } else if (const auto *funcDecl = dyn_cast<FunctionDecl>(decl)) {
     doFunctionDecl(funcDecl);
+  } else if (dyn_cast<HLSLBufferDecl>(decl)) {
+    llvm_unreachable("HLSLBufferDecl should not be handled here");
```

Comment:
No particular reason. The benefit is that we avoid another `dyn_cast<>`. Not a big deal I think.